### PR TITLE
Add Objective-C Lists/Arrays

### LINKED ISSUE #288: Add Objective-C Lists/Arrays/Hashes
# Description

Lists/Arrays/Hashes for Objective-C aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/lists.json` to the right Objective-C directory in `web/thesauruses/`
* Edit the lists.json file to accurately reflect the latest major version of Objective-C
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by vinayak0127**
## What GitHub issue does this PR apply to?

Resolves #288 

## Checklist

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at the documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

File: web/thesauruses/objectivec/lists.json [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+      "reduce_left",
+      "reduce_right"
+    ]
+  },
+  "lists": {
+    "name_of_ordered_mutable_list": {
+      "code": "NSMutableArray"
+    },
+    "create_a_ordered_mutable_list": {
+      "code": "NSMutable *fullName = [NSMutableArray arrWtihObj: @\"Jack\",@\"Sparrow\",nil]"
```
Comment (geekygirlsarah) [inline web/thesauruses/objectivec/lists.json:118]: "Jack Sparrow" 😂

File: web/thesauruses/objectivec/lists.json [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+    },
+    "ordered_mutable_list_can_be_inserted_in_middle": {
+      "code": "Yes",
+      "comment": "insert(_:at:)- Inserts a given object into the array’s contents at a given index."
+    },
+    "access_element_in_ordered_mutable_list": {
+      "code": "NSMutableArray arr = [Array1 objectAtIndex:i] objectIndex:j];"
+    },
+    "insert_into_beginning_of_ordered_mutable_list": {
+      "code": "insert(_:at:) - Inserts a given object into the array’s contents at a given index."
```
Comment (geekygirlsarah) [inline web/thesauruses/objectivec/lists.json]: I appreciate the definition, but the `code` line should be compileable code. Any comments or explanations should go in a `comment` line.

(Same for the lines below too)

File: web/thesauruses/objectivec/lists.json [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+    },
+    "unordered_mutable_list_start_number": {
+      "not-implemented": "true"
+    },
+    "unordered_mutable_list_can_be_appended": {
+      "code": "Yes"
+    },
+    "unordered_mutable_list_can_be_inserted_in_middle": {
+      "not-implemented": "true",
+      "comment": "addObject: - Adds a given object to the set, if it is not already a member."
```
Comment (geekygirlsarah) [inline web/thesauruses/objectivec/lists.json]: You have `not-implemented` here, but then you have a comment that sounds like this actually exists. Can you clarify? 

(You do this a few times below too)

File: web/thesauruses/objectivec/lists.json [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+    },
+    "erase_element_from_mutable_hashed_list": {
+      "code": "remove(_:)"
+    },
+    "delete_mutable_hashed_list": {
+      "code": "removeAllObjects()"
+    },
+    "name_of_immutable_hashed_list": {
+      "code": ""
+    },
```
Comment (geekygirlsarah) [inline web/thesauruses/objectivec/lists.json]: Is this not implemented? It should be set and the `code` line removed

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
There's a couple of questions and comments, but this is close to ready!

File: web/thesauruses/objectivec/lists.json (hunk 3 above)
Comment (vinayak0127) [reply web/thesauruses/objectivec/lists.json]: There is no actual function that is used to insert in the middle, addObject is used to insert at any index generally.
that's why I wrote that comment.

File: web/thesauruses/objectivec/lists.json (hunk 2 above)
Comment (vinayak0127) [reply web/thesauruses/objectivec/lists.json]: Okay, I'll keep that in mind.

File: web/thesauruses/objectivec/lists.json (hunk 1 above)
Comment (vinayak0127) [reply web/thesauruses/objectivec/lists.json:118]: Hey, should i change it ?😂

File: web/thesauruses/objectivec/lists.json (hunk 4 above)
Comment (vinayak0127) [reply web/thesauruses/objectivec/lists.json]: yes, I'll remove it.

File: web/thesauruses/objectivec/lists.json (hunk 1 above)
Comment (geekygirlsarah) [reply web/thesauruses/objectivec/lists.json:118]: Nah, I was amused!

File: web/thesauruses/objectivec/lists.json (hunk 2 above)
Comment (geekygirlsarah) [reply web/thesauruses/objectivec/lists.json]: Well, I was hoping you'd update them, or better yet, delete them. So `"code": "insert(_:at:)"` and `"comment": "Inserts a given object..."`  (or actually leave off `comment` all together). 

(See:  https://codethesauru-main-uzk96crhjcxu.herokuapp.com/reference/?concept=lists&lang=objectivec and how it's redundant.)

image

File: web/thesauruses/objectivec/lists.json (hunk 3 above)
Comment (geekygirlsarah) [reply web/thesauruses/objectivec/lists.json]: I guess to clarify on this more, this is a concept that's a question. CAN you insert something in the middle of a mutable list? So it should really just be yes/no, not the name of a function.

image

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
I think there's still a few more changes. You're close. If you can fix these things, I can merge it in.

**Comment by vinayak0127** [comment]
hey @geekygirlsarah  can you review new changes?

**Comment by geekygirlsarah** [comment]
Hi @vinayak0127. I am the only maintainer on this, and I also have a day job and side activities. I am assinging issues, answering comments, and reviewing and approving PRs, but I am busy and working on Code Thesaurus is not my main job. So I need patience as I am reviewing things in the order they come in. I will get to this, you just have to wait.

**Comment by geekygirlsarah** [comment]
Alright, I think the changes look good and it's ready to be merged.

Thank you for working on another issue on this and for making the suggested changes!

**Pull request merged**